### PR TITLE
ENACT-153: Update eslint-plugin-testing-library major dependency

### PR DESCRIPTION
**Pull request by andreitirlalge**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] I have run automated CLI testing and it is passed
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Updated `eslint-plugin-testing-library` major dependency.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
We are now using `eslint-plugin-testing-library` latest dependency. (v ^6.0.2)

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)
There were breaking changes affecting this repo.

### Links
[//]: # (Related issues, references)
ENACT-153
https://github.com/testing-library/eslint-plugin-testing-library/releases
https://github.com/testing-library/eslint-plugin-testing-library/blob/main/docs/migration-guides/v6.md

### Comments
Enact-DCO-1.0-Signed-off-by: Andrei Tirla andrei.tirla@lgepartner.com

### CODE REVIEW

**Review by ion-andrusciac-lgp — APPROVED** [review]
LGTM👍

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,15 @@
 
 The following is a curated list of changes in the Enact eslint config:
 
+## [unreleased] (October 5, 2023)
```
Comment (daniel-stoian-lgp) [inline CHANGELOG.md:5]:
```suggestion
## [unreleased]
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,15 @@
...
+
+* Removed deprecated rule `testing-library/no-wait-for-empty-callback`.
+
```
Comment (daniel-stoian-lgp) [inline CHANGELOG.md:8]:
```suggestion
* Removed deprecated rule `testing-library/no-wait-for-empty-callback`.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,15 @@
...
 The following is a curated list of changes in the Enact eslint config:
 
+## [unreleased]
+
+* Removed deprecated rule `testing-library/no-wait-for-empty-callback`.
+
+* Renamed `testing-library` rules for Jest unit tests validation:
+  * `testing-library/await-async-query` renamed to `testing-library/await-async-queries` 
+  * `testing-library/no-await-sync-query` renamed to `testing-library/no-await-sync-queries`
+  * `testing-library/no-render-in-setup` renamed to `testing-library/no-render-in-lifecycle`
```
Comment (daniel-stoian-lgp) [inline CHANGELOG.md:12]:
```suggestion
* Removed deprecated rule `testing-library/no-wait-for-empty-callback`.
* Renamed `testing-library` rules for Jest unit tests validation:
  * `testing-library/await-async-query` renamed to `testing-library/await-async-queries` 
  * `testing-library/no-await-sync-query` renamed to `testing-library/no-await-sync-queries`
  * `testing-library/no-render-in-setup` renamed to `testing-library/no-render-in-lifecycle`
```

**Review by daniel-stoian-lgp — APPROVED** [review]

**Review by hong6316 — APPROVED** [review]
👍

**Pull request merged**